security-analysis/src/main/java/eu/itesla_project/security/LimitViolation.java

```diff
@@ -32,8 +38,22 @@
 
     private final float baseVoltage;
 
+    @Deprecated
     public LimitViolation(Identifiable subject, LimitViolationType limitType, float limit, String limitName, float limitReduction, float value, Country country, float baseVoltage) {
         this.subject = Objects.requireNonNull(subject);
+        this.subjectId = subject.getId();
+        this.limitType = Objects.requireNonNull(limitType);
+        this.limit = limit;
+        this.limitName = limitName;
+        this.limitReduction = limitReduction;
+        this.value = value;
+        this.country = country;
+        this.baseVoltage = baseVoltage;
+    }
+    
+    public LimitViolation(String subjectId, LimitViolationType limitType, float limit, String limitName, float limitReduction, float value, Country country, float baseVoltage) {
+        this.subjectId = Objects.requireNonNull(subjectId);
+        this.subject = new IdentifiableImpl(subjectId); // to avoid null pointer exceptions on getSubject().getId()
```

Comment:
It's not a good ID to create a mock class to avoid null pointer. All existing getSubject().getId() statement will continue to work since everybody is using the first version of the constructor. The only requirement is the build of other repositories must not failed.
So you just have to write:
this.subject = null;
this.subjectId = Objects.requireNonNull(subjectId);